s/nub-keyboard/nub-keyboard.ts

```diff
@@ -0,0 +1,96 @@
+						<svg xmlns="http://www.w3.org/2000/svg" width="36" height="36" preserveAspectRatio="none" viewBox="0 0 24 24" fill="currentColor" stroke-width="2" class="ai ai-DragHorizontalFill"><path fill-rule="evenodd" clip-rule="evenodd" d="M22 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M22 16a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M14 16a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M6 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M6 16a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M14 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/></svg>
+						<svg xmlns="http://www.w3.org/2000/svg" width="36" height="36" preserveAspectRatio="none" viewBox="0 0 24 24" fill="currentColor" stroke-width="2" class="ai ai-DragHorizontalFill"><path fill-rule="evenodd" clip-rule="evenodd" d="M22 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M22 16a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M14 16a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M6 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M6 16a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/><path fill-rule="evenodd" clip-rule="evenodd" d="M14 8a2 2 0 1 1-4 0 2 2 0 0 1 4 0z"/></svg>
+					</div>
+				 	${Object.keys(keys).map((key: any, i) => `<button data-key="${key}" class="key">${key}</button>`).join("")}
+				</div>
+			</div>
+		`
+		const style = document.createElement("style")
+		style.textContent = styles
+		this.shadowRoot?.append(style)
+
+		const toggleEditor = this.shadowRoot?.querySelector<HTMLElement>(".toggle-editor")
+		const keysButtons = this.shadowRoot?.querySelectorAll<HTMLButtonElement>(".key")!
+		const draggableContainer: HTMLElement = this.shadowRoot?.querySelector(".draggable-item")!;
+		const editor = this.shadowRoot?.querySelector<HTMLElement>(".editor")!
+		const actionsElements = this.shadowRoot?.querySelectorAll<HTMLSelectElement>(".action")!
+		const itemToDrag:HTMLElement = document?.querySelector("nub-keyboard")!;
+
+		draggable(draggableContainer, itemToDrag)
+
+		toggleEditor?.addEventListener("pointerdown", () => {
```

Comment:
i think we should separate the editor out as its own component

`<nub-settings-bindings>`